test/limits/mzcompose.py

```diff
@@ -1238,8 +1238,14 @@ def body(cls) -> None:
 
             > INSERT INTO t1 VALUES (1);
 
-            #Increase SQL timeout to 10 minutes (~5 should be enough).
-            $ set-sql-timeout duration=600s
+            # Increase SQL timeout to 10 minutes (~5 should be enough).
```

Comment:
fyi @MaterializeInc/testing. The last three runs of this SELECT on main were 8m55s, 8m47s, 9m55s, so we're running pretty close to the 10m timeout already. I'm not sure why this setting would tickle it over the edge, but I spot checked on a fresh staging env and from 10 to 60, the timings are essentially the same with the setting on and off (I stopped after 60). The timings are definitely not linear though, so there may still be something for someone to look into here

Reply:
Wtf, I left this comment last week and github only now got around to sending the email out...